contrib/OtlpGrpc/SpanConverter.php

```diff
@@ -0,0 +1,108 @@
+        if (is_bool($value)) {
+            return $value ? 'true' : 'false';
+        }
+
+        // OTLP tags must be strings, but opentelemetry
+        // accepts strings, booleans, numbers, and lists of each.
+        if (is_array($value)) {
+            return join(',', array_map([$this, 'sanitiseTagValue'], $value));
+        }
+
+        // Floats will lose precision if their string representation
+        // is >=14 or >=17 digits, depending on PHP settings.
+        // Can also throw E_RECOVERABLE_ERROR if $value is an object
+        // without a __toString() method.
+        // This is possible because OpenTelemetry\Trace\Span does not verify
+        // setAttribute() $value input.
+        return (string) $value;
+    }
+
+    public function convert(Span $span)
+    {
+        $spanParent = $span->getParent();
+        $row = [
+            'id' => $span->getContext()->getSpanId(),
+            'traceId' => $span->getContext()->getTraceId(),
+            'parentId' => $spanParent ? $spanParent->getSpanId() : null,
+            'localEndpoint' => [
+                'serviceName' => $this->serviceName,
+            ],
+            'name' => $span->getSpanName(),
+            'timestamp' => (int) ($span->getStartEpochTimestamp() / 1e3), // RealtimeClock in microseconds
+            'duration' => (int) (($span->getEnd() - $span->getStart()) / 1e3), // Diff in microseconds
+
+        ];
+
+        foreach ($span->getAttributes() as $k => $v) {
+            if (!array_key_exists('tags', $row)) {
+                $row['tags'] = [];
+            }
+            $row['tags'][$k] = $this->sanitiseTagValue($v->getValue());
+        }
+
+        foreach ($span->getEvents() as $event) {
+            if (!array_key_exists('annotations', $row)) {
+                $row['annotations'] = [];
+            }
+            $row['annotations'][] = [
+                'timestamp' => (int) ($event->getTimestamp() / 1e3), // RealtimeClock in microseconds
+                'value' => $event->getName(),
+            ];
+        }
+
+        foreach ($span->getLinks() as $link) {
+            if (!array_key_exists('link', $row)) {
+                $row['link'] = [];
+            }
+            $row['link'][] = [
+                'trace_id' => $span->getContext()->getTraceId(),
+                'span_id' => $span->getContext()->getSpanId(),
+            ];
+        }
+        if (!array_key_exists('status', $row)) {
+                $proto_status = StatusCode::STATUS_CODE_OK;
+                //////
+                /////HIGHLIGHTED
+                ////
+                ///
+                if ($span->getStatus()->getCanonicalStatusCode() === "ERROR") {
+                    $proto_status = StatusCode::STATUS_CODE_ERROR;
+                }
+                $status=new Status();
+                $row['status']=$status->setCode('$proto_status')->setMessage("Description");
+            }
+
+        return $row;
+    }
+}
+            // $key=new keyValue();
```

Comment:
I think you left these comments in inadvertently.

Reply:
yup 😅